NumLib/Function/Interpolation.h

```diff
@@ -16,38 +16,69 @@
 namespace NumLib
 {
 
-/**
+namespace detail
+{
+
+template<unsigned DOFOffset, typename NodalValues, typename ShapeMatrix>
+void shapeFunctionInterpolate(
+        const NodalValues &/*nodal_values*/,
+        const ShapeMatrix &/*shape_matrix_N*/)
+{}
+
+template<unsigned DOFOffset, typename NodalValues, typename ShapeMatrix, typename... Numbers>
```

Comment:
Add documentation link to that function please.
Would ScalarType reflect the semantic of that type?